packages/perseus-editor/src/widgets/interactive-graph-editor/locked-figures/locked-line-settings.tsx

```diff
@@ -73,13 +74,34 @@ const LockedLineSettings = (props: Props) => {
 
     // Check if the line has length 0.
     const isInvalid = kvector.equal(point1.coord, point2.coord);
-
-    function getPrepopulatedAriaLabel() {
+    /**
+     * Generate a prepopulated aria label for the line, with the math
+     * details converted into spoken words.
+     */
+    async function getPrepopulatedAriaLabel() {
         let visiblelabel = "";
+        let point1VisibleLabel = "";
+        let point2VisibleLabel = "";
+
         if (labels && labels.length > 0) {
             visiblelabel += ` ${labels.map((l) => l.text).join(", ")}`;
         }
-        let str = `${capitalizeKind}${visiblelabel} from (${point1.coord[0]}, ${point1.coord[1]}) to (${point2.coord[0]}, ${point2.coord[1]})`;
+
+        if (point1.labels && point1.labels.length > 0) {
+            point1VisibleLabel += ` ${point1.labels
+                .map((l) => l.text)
+                .join(", ")}`;
+        }
+
+        if (point2.labels && point2.labels.length > 0) {
+            point2VisibleLabel += ` ${point2.labels
+                .map((l) => l.text)
+                .join(", ")}`;
+        }
+
+        let str = await generateSpokenMathDetails(
+            `${capitalizeKind}${visiblelabel} from point${point1VisibleLabel} at (${point1.coord[0]}, ${point1.coord[1]}) to point${point2VisibleLabel} at (${point2.coord[0]}, ${point2.coord[1]})`,
+        );
```

Comment:
The duplication of the `labels.map((l) => l.text).join(", ")` logic makes me think there's a missing abstraction here. What exactly that abstraction should be probably depends on how labels are being constructed for other locked figure types. It would be nice if we could represent this logic in one place.

I'm imagining something like:

```ts
function describeLockedPoint(point: LockedPointType): string {
  const {coord, labels} = point;
  const visibleLabel = labels && labels.length > 0
    ? ` ${labels.map((l) => l.text).join(", ")}`
    : "";

  return `point${visibleLabel} at (${coord[0]}, ${coord[1]})`
}
```

but I'm not sure this would be suitable for all locked figures.

Reply:
Alternatively, maybe we could have a function like `joinLockedFigureLabels` that takes in an array of LockedLabels and then gets used within here. I thought about abstracting away the whole description, but the description does change a bit for every figure, which is why I didn't end up going that way.

Reply:
I would definitely be for creating a function for this. Especially since we could potentially re-use it in future implementations. Plus we can make calls to `joinLockedFigureLabels` and add a comment describing the string it's generating. 😊